### PR TITLE
create beaufort cipher

### PR DESCRIPTION
**Pull request by radadiyamohit81**
if you like my code, merge it and add the label as `hacktoberfest-accepted`

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
Please run your code thru psf/black as discussed in CONTRIBUTING.md.

**Comment by cclauss** [comment]
Here we need single quotes... https://travis-ci.com/github/TheAlgorithms/Python/builds/190633967#L782-L804

File: ciphers/beaufort_cipher.py [hunk 1]
```diff
@@ -66,7 +66,7 @@
 # equal to the length of original text
 def generate_key(message: str, key: str) -> str:
     """
-    >>> generate_key("THE GERMAN ATTACK","SECRET")
+    >>> generate_key('THE GERMAN ATTACK','SECRET')
     SECRETSECRETSECRE
```
Comment (cclauss) [inline ciphers/beaufort_cipher.py]:
```suggestion
    >>> generate_key("THE GERMAN ATTACK", "SECRET")
    'SECRETSECRETSECRE'
```

**Comment by radadiyamohit81** [comment]
Thank you @cclauss  for helping me Thank you so much

File: ciphers/beaufort_cipher.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    9: "J",
+    10: "K",
+    11: "L",
+    12: "M",
+    13: "N",
+    14: "O",
+    15: "P",
+    16: "Q",
+    17: "R",
+    18: "S",
```
Comment (cclauss) [inline ciphers/beaufort_cipher.py:53]:
```suggestion
from string import ascii_uppercase

dict1 = {char: i for i, char in enumerate(ascii_uppercase)}
dict2 = {i: char for i, char in enumerate(ascii_uppercase)}
```
Dict comprehensions!

File: ciphers/beaufort_cipher.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        if len(key) == len(message):
+            break
+        key += key[i]
+        i += 1
+    return key
+
+
+# This function returns the encrypted text
+# generated with the help of the key
+def cipherText(message: str, key_new: str) -> str:
```
Comment (cclauss) [inline ciphers/beaufort_cipher.py:86]:
```suggestion
def cipher_text(message: str, key_new: str) -> str:
```

File: ciphers/beaufort_cipher.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        else:
+            x = (dict1[letter] - dict1[key_new[i]]) % 26
+            i += 1
+            cipher_text += dict2[x]
+    return cipher_text
+
+
+# This function decrypts the encrypted text
+# and returns the original text
+def originalText(cipher_text: str, key_new: str) -> str:
```
Comment (cclauss) [inline ciphers/beaufort_cipher.py:105]:
```suggestion
def original_text(cipher_text: str, key_new: str) -> str:
```

File: ciphers/beaufort_cipher.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+
+def main():
+    message = "THE GERMAN ATTACK"
+    key = "SECRET"
+    key_new = generate_key(message, key)
+    cipher_text = cipherText(message, key_new)
+    original_text = originalText(cipher_text, key_new)
+    print("Encrypted Text =", cipher_text)
+    print("Original Text =", original_text)
```
Comment (cclauss) [inline ciphers/beaufort_cipher.py:129]:
```suggestion
    s = cipher_text(message, key_new)
    print(f"Encrypted Text = {s}")
    print(f"Original Text = {original_text(s, key_new)}")
```
f-strings...

File: ciphers/beaufort_cipher.py [hunk 6]
```diff
@@ -102,7 +49,7 @@ def cipherText(message: str, key_new: str) -> str:
 
 # This function decrypts the encrypted text
 # and returns the original text
-def originalText(cipher_text: str, key_new: str) -> str:
+def original_text(cipher_text: str, key_new: str) -> str:
     """
     >>> originalText("BDC PAYUWL JPAIYI","SECRETSECRETSECRE")
```
Comment (cclauss) [inline ciphers/beaufort_cipher.py:54]:
```suggestion
    >>> original_text("BDC PAYUWL JPAIYI","SECRETSECRETSECRE")
```

**Review by cclauss — APPROVED** [review]
Nice one!  Thanks for your persistance.

**Comment by radadiyamohit81** [comment]
Thank you so much @cclauss, looking forward to contributing more to your repository.

**Comment by radadiyamohit81** [comment]
I am also trying to contribute a new algorithm related to cipher #3449

**Pull request merged**